### PR TITLE
tests: Fix integration tests

### PR DESCRIPTION
**Pull request by stepansergeevitch**
Fix integration tests for fb1.0 and metabase 48

### CODE REVIEW

**Comment by stepansergeevitch** [comment]
Fixed some of the issues in metabase driver integration tests. Still a couple left . I'll create JIRA tickets for them so we are able to prioritize them and fix by demand

File: src/metabase/driver/firebolt.clj [hunk 1]
```diff
@@ -101,64 +101,61 @@
...
+(defmethod sql.qp/date [:firebolt :quarter-of-year] [_ _ expr] (extract "quarter" expr))
 
 ; Set start of week to be monday
 (defmethod driver/db-start-of-week :firebolt
   [_]
   :monday)
 
 ; Modify start of week to monday
-(defn- to-start-of-week [expr] [:'toMonday expr])
+(defn- to-start-of-week [expr](extract "week" expr))
```
Comment (ptiurin) [inline src/metabase/driver/firebolt.clj]: I think this is different. Clickhouse `toMonday` returns the date of the nearest Monday, while we return the current week.

File: src/metabase/driver/firebolt.clj [hunk 2]
```diff
@@ -172,31 +169,43 @@
...
             (let [hsql-field (sql.qp/->honeysql :firebolt field)
                   hsql-value (sql.qp/->honeysql :firebolt value)]
                     (if (get options :case-sensitive true)
-                           [:> [:'positionUTF8                hsql-field hsql-value] 0]
-                           [:> [:'positionCaseInsensitiveUTF8 hsql-field hsql-value] 0])))
+                      [:nest [:> [:STRPOS hsql-field hsql-value] 0]]
+                      [:nest [:> [:STRPOS [:LOWER hsql-field] [:LOWER hsql-value]] 0]])))
+
+(defn escape-regexp-sql [clause]
+  [:REGEXP_REPLACE_ALL clause "([!$()*+.:<=>?[\\\\\\]^{|}-])" "\\\\\\\\\\\\1"])
```
Comment (ptiurin) [inline src/metabase/driver/firebolt.clj:187]: Let's add a comment what this does.

File: src/metabase/driver/firebolt.clj [hunk 3]
```diff
@@ -333,28 +342,24 @@
...
+(defmethod driver/database-supports? [:firebolt :regex]  [_ _ _] true)
 
-(defmethod driver/database-supports? [:firebolt :nested-fields]  [_ _] false)
+(defmethod driver/database-supports? [:firebolt :standard-deviation-aggregations]  [_ _ _] false)
 
+(defmethod driver/database-supports? [:firebolt :nested-queries]  [_ _ _] false)
 
+(defmethod driver/database-supports? [:firebolt :case-sensitivity-string-filter-options]  [_ _ _] false)
 
+(defmethod driver/database-supports? [:firebolt :set-timezone]  [_ _ _] false)
```
Comment (ptiurin) [inline src/metabase/driver/firebolt.clj:364]: I think this is supported, right?

File: src/metabase/driver/firebolt.clj [hunk 4]
```diff
@@ -333,28 +342,24 @@
...
-(defmethod driver/database-supports? [:firebolt :case-sensitivity-string-filter-options]  [_ _] false)
+(defmethod driver/database-supports? [:firebolt :binning]  [_ _ _] true)
 
-(defmethod driver/database-supports? [:firebolt :set-timezone]  [_ _] false)
+(defmethod driver/database-supports? [:firebolt :regex]  [_ _ _] true)
 
-(defmethod driver/database-supports? [:firebolt :nested-fields]  [_ _] false)
+(defmethod driver/database-supports? [:firebolt :standard-deviation-aggregations]  [_ _ _] false)
 
+(defmethod driver/database-supports? [:firebolt :nested-queries]  [_ _ _] false)
```
Comment (ptiurin) [inline src/metabase/driver/firebolt.clj:360]: Also this is true?

**Review by ptiurin — COMMENTED** [review]
Some comments with my limited understanding of Clojure.

File: src/metabase/driver/firebolt.clj (hunk 3 above)
Comment (stepansergeevitch) [reply src/metabase/driver/firebolt.clj:364]: This needs some additional implementation to make it enabled. We can do this as a part of a separate task

File: src/metabase/driver/firebolt.clj (hunk 1 above)
Comment (stepansergeevitch) [reply src/metabase/driver/firebolt.clj]: Good point, we should use `date-trunc` here

File: src/metabase/driver/firebolt.clj (hunk 4 above)
Comment (stepansergeevitch) [reply src/metabase/driver/firebolt.clj:360]: This needs to be validated serapaterly as well. We support nested queries in FB, but we might also need to adjust driver implementation

File: src/metabase/driver/firebolt.clj (hunk 3 above)
Comment (stepansergeevitch) [reply src/metabase/driver/firebolt.clj:364]: https://packboard.atlassian.net/browse/FIR-35123

File: src/metabase/driver/firebolt.clj (hunk 4 above)
Comment (stepansergeevitch) [reply src/metabase/driver/firebolt.clj:360]: https://packboard.atlassian.net/browse/FIR-35124

**Comment by stepansergeevitch** [comment]
A follow up jira to fix remaining tests: https://packboard.atlassian.net/browse/FIR-35102

**Review by ptiurin — APPROVED** [review]

**Pull request merged**